Proiect_PAO/src/aplicatieBancara/Card/Card.java

```diff
@@ -0,0 +1,108 @@
+package aplicatieBancara.Card;
+
+import aplicatieBancara.Cont.Cont;
+import aplicatieBancara.TipCard;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+public abstract class Card {
+
+    protected final int cardId;
+    protected final int CVV;
+    protected String numar;
+    protected String IBAN;
+    protected double sold;
+    protected Cont cont;
+    protected final Date dataExpirare;
+    //protected String PIN;
+
+
+    static private final HashSet<String> usedNumbers = new HashSet<String>();
+
+    abstract public String getTip();
+
+    public Card(int cardId, Cont cont) {
+        this.cardId = cardId;
+        this.IBAN = cont.getIBAN();
+        this.cont = cont;
+        this.numar = this.generareNumarCard();
+
+        //generare numar card
+        while(usedNumbers.contains(this.numar))
+            this.numar = this.generareNumarCard();
+        usedNumbers.add(this.numar);
+
+        //generare CVV
+        this.CVV = this.generareCVV();
+
+        //generare data expirare
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.YEAR, 3);
+        this.dataExpirare = c.getTime();
+    }
+
+
+    public double interogareSold()
+    {
+        return this.cont.interogareSold();
+    }
+
+    @Override
+    public String toString() {
+        return  "cardId: " + cardId + "\n" +
+                "CVV: " + CVV + "\n" +
+                "Numar: " + numar + "\n" +
+                "Nume titular: " + this.cont.getNumeTitular() + "\n" +
+                "IBAN: " + IBAN + "\n" +
+                "Data expirare: " + dataExpirare + "\n";
+    }
+
+    private String generareNumarCard() {
```

Comment:
can be static and in constructor referenced without this, since it is just a helper method